app/src/main/java/com/keylesspalace/tusky/TuskyApplication.kt

```diff
@@ -91,4 +99,18 @@ class TuskyApplication : Application(), HasAndroidInjector {
     }
 
     override fun androidInjector() = androidInjector
+
+    private fun upgradeSharedPreferences(oldVersion: Int, newVersion: Int) {
```

Comment:
I was wondering about this task of preference migration handling over there: https://github.com/tuskyapp/Tusky/pull/3205

Especially I was wondering if there is already a sort of migrations (for the database only) and if such a migration should be also located there.

Both task are "update the stored files from one version to the next".

Reply:
Eventually I'd like anything that looks like a preference (whether that's something that's an actual Android Preference subclass, or some other setting that the user interacts with but is stored differently) to be accessed via a new `PreferenceRepository`.

This would provide type-safe access to the preferences, and all the storage / migration logic can be stored in one place. I go through some of this in https://github.com/tuskyapp/Tusky/pull/2393#issuecomment-1374580034.

I think trying to do all that once is too big a change, so I'm thinking about incremental PRs that move closer to the goal.

Reply:
However I wouldn't see the normal db content (i.e. status) not as a preference.

(Regarding my question if "all migration should be handled similarly".)